java-checks/src/main/java/org/sonar/java/checks/methods/MethodInvocationMatcher.java

```diff
@@ -97,19 +98,24 @@ public MethodInvocationMatcher withNoParameterConstraint() {
 
   public boolean matches(NewClassTree newClassTree) {
     NewClassTreeImpl newClassTreeImpl = (NewClassTreeImpl) newClassTree;
-    return matches(newClassTreeImpl.getConstructorIdentifier(), null);
+    return matches(newClassTreeImpl.getConstructorIdentifier().symbol(), null);
   }
 
   public boolean matches(MethodInvocationTree mit) {
     IdentifierTree id = getIdentifier(mit);
-    if (id != null) {
-      return matches(id, getCallSiteType(mit));
+    return id != null && matches(id.symbol(), getCallSiteType(mit));
+  }
+
+  public boolean matches(MethodTree methodTree) {
+    MethodSymbol symbol = methodTree.symbol();
+    if (symbol != null) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Redundant nullcheck of symbol, which is known to be non-null in org.sonar.java.checks.methods.MethodInvocationMatcher.matches(MethodTree) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)